lib/wallaroo/core/sink/connector_sink/connector_sink_notify.pony

```diff
@@ -0,0 +1,455 @@
+/*
+
+Copyright 2019 The Wallaroo Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ implied. See the License for the specific language governing
+ permissions and limitations under the License.
+
+*/
+
+use "backpressure"
+use "buffered"
+use "net"
+use "wallaroo/core/common"
+use "wallaroo/core/network"
+use "wallaroo_labs/bytes"
+use cp = "wallaroo_labs/connector_protocol"
+use "wallaroo_labs/mort"
+
+class ConnectorSinkNotify
+  var _fsm_state: cp.ConnectorProtoFsmState = cp.ConnectorProtoFsmDisconnected
+  var _header: Bool = true
+  var _connected: Bool = false
+  var _throttled: Bool = false
+  let _stream_id: cp.StreamId = 1
+  let _sink_id: RoutingId
+  let _worker_name: WorkerName
+  let _protocol_version: String
+  let _cookie: String
+  let _auth: ApplyReleaseBackpressureAuth
+  let stream_name: String
+  var credits: U32 = 0
+  var acked_point_of_ref: cp.MessageId = 0
+  var message_id: cp.MessageId = acked_point_of_ref
+  var _connection_count: USize = 0
+  // 2PC
+  var _rtag: U64 = 77777
```

Comment:
As we're considering extracing the 2PC component of the sink, could these bits be encapsulated and extracted out as well?